### PR TITLE
Use unbounded allocation before returning.

### PR DESCRIPTION
**Pull request by ekpyron**
Reverts already do things like that.

### CODE REVIEW

File: libsolidity/codegen/ir/IRGenerator.cpp [hunk 1]
```diff
@@ -870,7 +870,7 @@ string IRGenerator::dispatchRoutine(ContractDefinition const& _contract)
 				<callValueCheck>
 				<?+params>let <params> := </+params> <abiDecode>(4, calldatasize())
 				<?+retParams>let <retParams> := </+retParams> <function>(<params>)
-				let memPos := <allocate>(0)
+				let memPos := <allocate>()
```
Comment (chriseth) [inline libsolidity/codegen/ir/IRGenerator.cpp]:
```suggestion
				let memPos := <allocateUnbounded>()
```
?

File: test/libsolidity/semanticTests/abiEncoderV1/abi_decode_v2_storage.sol [hunk 2]
```diff
@@ -24,6 +24,6 @@ contract C {
 // compileViaYul: also
 // ----
 // f() -> 0x20, 0x8, 0x40, 0x3, 0x9, 0xa, 0xb
-// gas irOptimized: 193697
+// gas irOptimized: 193626
```
Comment (bshastry) [inline test/libsolidity/semanticTests/abiEncoderV1/abi_decode_v2_storage.sol:27]: Nice! :-)

File: libsolidity/codegen/ir/IRGenerator.cpp (hunk 1 above)
Comment (ekpyron) [inline libsolidity/codegen/ir/IRGenerator.cpp]: Funny, by the way, that it didn't actually allocate the proper size anyways, but always just zero :-).

Comment (ekpyron) [reply libsolidity/codegen/ir/IRGenerator.cpp]: Alright, changed.

**Review by chriseth — DISMISSED** [review]

**Review by bshastry — COMMENTED** [review]
Could you please add a test case e.g., one where free mem ptr is corrupted and `compileViaYul: also` leading to no return but successful call?

File: test/libsolidity/semanticTests/abiEncoderV2/abi_encode_v2_in_modifier_used_in_v1_contract.sol [hunk 3]
```diff
@@ -38,5 +38,5 @@ contract C is B {
 // compileViaYul: also
 // ----
 // test() -> 5, 10
-// gas irOptimized: 101019
+// gas irOptimized: 92642
```
Comment (hrkrshnn) [inline test/libsolidity/semanticTests/abiEncoderV2/abi_encode_v2_in_modifier_used_in_v1_contract.sol:41]: Saves 8k gas? Very surprising.

Comment (hrkrshnn) [reply test/libsolidity/semanticTests/abiEncoderV2/abi_encode_v2_in_modifier_used_in_v1_contract.sol:41]: Maybe the function call got inlined? Can you check?

Comment (ekpyron) [reply test/libsolidity/semanticTests/abiEncoderV2/abi_encode_v2_in_modifier_used_in_v1_contract.sol:41]: It's code size - the modifier creates a contract and this change saves quite a bit of code.

**Comment by ekpyron** [comment]
> Could you please add a test case e.g., one where free mem ptr is corrupted and `compileViaYul: also` leading to no return but successful call?

Added a test case - even though I'm not entirely sure how reasonable it is to test this, since it still stays undefined behaviour :-).

**Review by chriseth — APPROVED** [review]

**Pull request merged**